src/main/scala/io/iohk/ethereum/vm/VM.scala

```diff
@@ -115,8 +115,8 @@ class VM[W <: WorldStateProxy[W, S], S <: Storage[S]] extends Logger {
     ProgramResult(ByteString.empty, context.startGas, context.world, Set(), Nil, Nil, 0, Some(InvalidCall))
 
   private def saveNewContract(context: PC, address: Address, result: PR, config: EvmConfig): PR = {
-    if(result.error.contains(RevertOccurs)) {
-      result
+    if(result.error.isDefined) {
+      if (result.error.contains(RevertOccurs)) result else result.copy(gasRemaining = 0)
```

Comment:
sorry, my bad i rushed with merge :) 

I based it on https://github.com/ethereum/go-ethereum/blob/223fe3f26e8ec7133ed1d7ed3d460c8fc86ef9f8/core/vm/evm.go#L378, where they use all gas if error is other than revert.

Reply:
Ok, I wasn't aware of that. Please correct me if I wrong, don't we need to check if the problem occurs in block >= homestead?

Reply:
If you decipher tis expression`(err != nil && (evm.ChainConfig().IsHomestead(evm.BlockNumber) || err != ErrCodeStoreOutOfGas))` you will see that if we have any error different than `ErrCodeStoreOutOfGas` it will evaluate to true even not for homestead, homestead is only needed for `ErrCodeStoreOutOfGas` 
which we are covering in next if clauses:
https://github.com/input-output-hk/mantis/pull/553/files#diff-5b7d8817413ce869c362b5398e9a65f1R127